tuf/download.py

```diff
@@ -257,14 +231,57 @@ def _download_file(url, required_length, STRICT_REQUIRED_LENGTH=True):
   temp_file = securesystemslib.util.TempFile()
 
   try:
-    # Open the connection to the remote file.  _open_connection() can raise
-    # socket connection exceptions (such as SSLError).  Connection errors of
-    # this kind can be minimized by adjusting the socket timeout in
-    # settings.py.
-    connection = _open_connection(url)
+    # Use a different requests.Session per hostname to reuse connections while
+    # minimizing subtle security issues.
+    hostname = six.moves.urllib.parse.urlparse(url).hostname
+
+    if not hostname:
+        raise securesystemslib.exceptions\
+                            .FormatError('Could not get hostname from: ' + url)
+    else:
+        logger.debug('url: ' + url)
+        logger.debug('hostname: ' + hostname)
+
+    session = _sessions.get(hostname)
+
+    if not session:
+        session = requests.Session()
+        _sessions[hostname] = session
+
+        # Attach some default headers to every Session.
+        requests_user_agent = session.headers['User-Agent']
+        # Follows the RFC: https://tools.ietf.org/html/rfc7231#section-5.5.3
+        tuf_user_agent = 'tuf/' + tuf.__version__ + ' ' + requests_user_agent
+        session.headers.update({
+            # Tell the server not to compress or modify anything.
+            # https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Encoding#Directives
+            'Accept-Encoding': 'identity',
+            # The TUF user agent.
+            'User-Agent': tuf_user_agent
+        })
+
+        logger.debug('Made new session for ' + hostname)
+    else:
+        logger.debug('Reusing session for ' + hostname)
+
+    # Get the requests.Response object for this URL.
+    #
+    # Always stream to control how requests are downloaded:
+    # http://docs.python-requests.org/en/master/user/advanced/#body-content-workflow
+    #
+    # We will always manually close Responses, so no need for a context
+    # manager.
+    #
+    # Always set the connect + read timeout:
+    # http://docs.python-requests.org/en/master/user/advanced/#timeouts
+    response = session.get(url, stream=True,
+                           timeout=tuf.settings.SOCKET_TIMEOUT)
+
+    # Check response status.
+    response.raise_for_status()
```

Comment:
Not style compatible, so fixed:
 - indent size: 4 if continuing line else 2
 - line arrangement per PEP 8